src/Compilers/Core/Portable/Compilation/IExpression.cs

```diff
@@ -841,10 +841,36 @@ public interface IConversionExpression : IHasOperatorMethodExpression
         /// Kind of conversion.
         /// </summary>
         ConversionKind ConversionKind { get; }
+
+        /// <summary>
+        /// Returns true if the conversion is an identity conversion.
+        /// </summary>
+        bool IsIdentity { get; }
+
+        /// <summary>
+        /// Returns true if the conversion is an implicit ('widening' in VB) conversion.
+        /// </summary>
+        bool IsImplicit { get; }
```

Comment:
Unfortunately, the C# naming sort-of implies that the cast was in source or not. Typically implicit conversion operators do not need to be mentioned in source, but explicit ones do. Hence the names. However, for example, being an instance of an implicit conversion does not actually tell you whether it was actually in source or not. So if we use the implicit/explicit terms we tend to confuse ourselves all the time about what we mean.  "Was this cast implicit?"  What question is being asked?